pkg/kbuffer/dedupe.go

```diff
@@ -31,23 +35,34 @@ var shuttingDownError = ActivationResult{
 var _ Activator = (*dedupingActivator)(nil)
 
 type dedupingActivator struct {
-	mux             sync.Mutex
-	pendingRequests map[revisionID][]chan ActivationResult
-	activator       Activator
-	shutdown        bool
+	mux              sync.Mutex
+	pendingRequests  map[revisionID][]chan ActivationResult
+	activator        Activator
+	endpointObserver EndpointObserver
+	kubeClient       kubernetes.Interface
+	throttler Throttler
```

Comment:
Nit: spacing.  I suggest running `gofmt` on every save.

Reply:
yeap, had  it before in Goland, got lost somewhere